spec/controllers/devise/two_factor_authentication_controller_spec.rb

```diff
@@ -71,4 +78,37 @@
       end
     end
   end
+
+  describe '#new' do
+    let(:user) { create(:user, :signed_up) }
+
+    context 'when not yet fully authenticated' do
+      before do
+        sign_in_before_2fa
+      end
+
+      it 'redirects to :show' do
+        get :new
+        expect(response).to redirect_to(action: :show)
+      end
+
+      it 'sends a new OTP' do
+        old_otp = subject.current_user.direct_otp
+        get :new
+        expect(SmsSenderOtpJob).to_not receive(:perform_later).with(user)
```

Comment:
minor style nit: please add an empty line above the expectations to separate them from the rest of the test.

Also, expectations that check if an object has received a message need to be placed before the action. So, line 98 needs to go before line 97 in this test. And in this case, the expectation is that `SmsSenderOtpJob` *should* receive `perform_later` since we are expecting a new OTP to be sent.